contracts/MemeBase.sol

```diff
@@ -91,7 +109,29 @@ contract MemeBase is MemeFactory {
         balancerPoolId = _balancerPoolId;
 
         if (accounts.length > 0) {
-            _redemptionSetup(accounts, amounts);
+            redemptionMemeBase = _redemptionMemeBase;
+
+            // Old AGNT token is the 4th in the global list of old MemeBase tokens
+            // https://basescan.org/address/0x42156841253f428cb644ea1230d4fddfb70f8891#readContract#F18
+            // Previous token address: 0x7484a9fB40b16c4DFE9195Da399e808aa45E9BB9
+            address redemptionMemeToken = IMemeBase(_redemptionMemeBase).memeTokens(_redemptionTokenIdx);
+
+            // Full collected AGNT amount: 141569842100000000000
+            // https://basescan.org/address/0x42156841253f428cb644ea1230d4fddfb70f8891#readContract#F17
+            MemeFactory.MemeSummon memory memeSummon = IMemeBase(_redemptionMemeBase).memeSummons(redemptionMemeToken);
+            contributionAGNT = memeSummon.nativeTokenContributed;
+
+            // Redemption amount: collected amount - 10% for burn = 127412857890000000000
+            redemptionAmount = contributionAGNT * 9 / 10;
+
+            // Set up redemption procedure
+            uint8 decimals = IERC20(redemptionMemeToken).decimals();
+            // Half of the supply has been burnt, thus it needs to be increased
+            uint256 totalSupply = 2 * IERC20(redemptionMemeToken).totalSupply();
```

Comment:
This is not safe to do. This was really not useful work. What if someone buys and burns tokens now. Just hardcode it